### PR TITLE
Allow provisioning of a generic bucket

### PR DESCRIPTION
**Pull request by zaro0508**
Make this script default to provision a generic bucket.  A generic
bucket is a bucket that does not have any policies applied to it.
An optional parameter is provided to make it a Synapse Custom bucket.

### CODE REVIEW

File: templates/managed-s3.yaml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  SynapseBucket:
+    Type: String
+    Description: true for a synapse bucket, false (default) for general bucket
+    AllowedValues:
+      - true
+      - false
+    Default: false
+  EncryptBucket:
+    Type: String
+    Description: true (default) to encrypt bucket, false for no encryption
```
Comment (thomasyu888) [inline templates/managed-s3.yaml]: Should this always just be true. No false allowed?

File: templates/managed-s3.yaml [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  BucketVersioning:
+    Type: String
+    Description: Enabled to enable bucket versionsing, default is Suspended
+    AllowedValues:
+      - Enabled
+      - Suspended
+    Default: Suspended
+  SynapseUserName:
+    Type: String
+    Description: Synapse user for read-write bucket
```
Comment (thomasyu888) [inline templates/managed-s3.yaml]: Is this only relevant if synapse bucket is set to true? Furthermore, can you specify a team?

File: templates/managed-s3.yaml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    Default: GLACIER
+  LifecycleDataExpiration:
+    Type: Number
+    Description: Number of days (from creation) when objects are deleted from S3 and the LifecycleDataStorageClass
+    Default: 365000
+Conditions:
+  AllowWrite: !Equals [!Ref AllowWriteBucket, true]
+  EnableSynapseBucket: !Equals [!Ref SynapseBucket, true]
+  EnableEncryption: !Equals [!Ref EncryptBucket, true]
+  DisableEncryption: !Not [!Condition EnableEncryption]
```
Comment (thomasyu888) [inline templates/managed-s3.yaml]: Oh I see. Does this part require the bucket to be encrypted?

File: templates/managed-s3.yaml (hunk 1 above)
Comment (zaro0508) [reply templates/managed-s3.yaml]: well, on scicomp should be true always but on sandbox could go either way?

File: templates/managed-s3.yaml (hunk 2 above)
Comment (zaro0508) [reply templates/managed-s3.yaml]: This is the synapse username, so it's whatever you used when registering with synapse. 
 I don't think it supports anything else at least i don't see anything else supported in the synapse docs

File: templates/managed-s3.yaml (hunk 3 above)
Comment (zaro0508) [reply templates/managed-s3.yaml]: no, there is no requirement to be encrypted.  this is just the inverse logic for CF templates.

**Comment by zaro0508** [comment]
Yes, I've never used a synapse bucket but I believe your description is accurate.

**Comment by zaro0508** [comment]
No, it's not a lambda.  We use the sceptre-sage-provisioner s3_notify hook[1] to read the SynapseUserName from the CF outputs and create the owner.txt file.

[1] https://github.com/Sage-Bionetworks/sceptre-provisioner-hooks/pull/5

**Comment by zaro0508** [comment]
This is old.  Doesn't seem very important either.  Scientists typically want synapse buckets.

**Pull request closed without merging**